nuxeo-features/nuxeo-platform-rendition/nuxeo-platform-rendition-core/src/test/java/org/nuxeo/ecm/platform/rendition/service/lazy/TestDeliverStaleLazyRenditions.java

```diff
@@ -0,0 +1,222 @@
+            throw new NuxeoException(e);
+        }
+    }
+
+    protected void checkStoredStaleRendition(int count, String latestRenditionDigest) {
+        try {
+            log.debug(String.format("Check stale rendition #%d", count));
+
+            log.debug("Update dc:issued on test document");
+            Thread.sleep(1000);
+            doc.setPropertyValue("dc:issued", new GregorianCalendar());
+            session.saveDocument(doc);
+            txFeature.nextTransaction();
+
+            log.debug("Ask immediately for a lazy rendition, expecting always the same stored stale rendition");
+            checkStaleRendition(doc, "lazyAutomation", true, "testDoc.txt", latestRenditionDigest);
+
+            log.debug("Wait for async completion");
+            waitForAsyncCompletion();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new NuxeoException(e);
+        }
+    }
+
+    protected void checkUpToDateRendition(boolean store) {
+        log.debug("Ask for a lazy rendition, should be rendered, expecting an up-to-date rendition");
+        checkUpToDateRendition(doc, "lazyAutomation", store, "testDoc.txt", DummyDocToTxt.getDigest(doc));
+    }
+
+    protected void checkEmptyRendition(DocumentModel doc, String renditionName, boolean store) {
+        Rendition rendition = rs.getRendition(doc, renditionName, store);
+        assertNotNull(rendition);
+        assertFalse(rendition.isStored());
+        Blob renditionBlob = rendition.getBlob();
+        assertTrue(renditionBlob.getMimeType().contains(LazyRendition.EMPTY_MARKER));
+        assertTrue(renditionBlob.getFilename().equals(LazyRendition.IN_PROGRESS_MARKER));
+        assertEquals(0, renditionBlob.getLength());
+    }
+
+    protected void checkStaleRendition(DocumentModel doc, String renditionName, boolean store, String expectedFilename,
+            String expectedDigest) {
+        checkRenderedRendition(doc, renditionName, store, true, expectedFilename, expectedDigest);
+    }
+
+    protected void checkUpToDateRendition(DocumentModel doc, String renditionName, boolean store,
+            String expectedFilename, String expectedDigest) {
+        checkRenderedRendition(doc, renditionName, store, false, expectedFilename, expectedDigest);
+    }
+
+    protected void checkRenderedRendition(DocumentModel doc, String renditionName, boolean store, boolean stale,
+            String expectedFilename, String expectedDigest) {
+        Rendition rendition = rs.getRendition(doc, renditionName, store);
+        assertNotNull(rendition);
+        assertTrue(rendition.isStored() == store);
+        Blob blob = rendition.getBlob();
+        String mimeType = blob.getMimeType();
+        assertFalse(blob.getMimeType().contains(LazyRendition.EMPTY_MARKER));
+        assertTrue(
+                stale ? mimeType.contains(LazyRendition.STALE_MARKER) : !mimeType.contains(LazyRendition.STALE_MARKER));
```

Comment:
You could use `assertTrue(stale == mimeType.contains(LazyRendition.STALE_MARKER))`

Reply:
:+1: